DuckDuckGo/HomePage/Model/HomePageSettings/UserBackgroundImagesManager.swift

```diff
@@ -222,6 +235,8 @@ final class UserBackgroundImagesManager: UserBackgroundImagesManaging {
     private func deleteImageFiles(for userBackgroundImage: UserBackgroundImage) {
         FileManager.default.remove(fileAtURL: storageLocation.appendingPathComponent(userBackgroundImage.fileName))
         FileManager.default.remove(fileAtURL: thumbnailsStorageLocation.appendingPathComponent(userBackgroundImage.fileName))
+        imagesCache.removeValue(forKey: userBackgroundImage)
+        thumbnailsCache.removeValue(forKey: userBackgroundImage)
```

Comment:
Here the cache is cleared upon image deletion.

It can't easily be unit-tested without abstracting away the `NSImage(contentsOfFile:)` initializer, but there's a unit test called `testWhenDeleteImageIsPassedExistingImageThenImageIsRemovedAndFilesAreDeleted` that checks that `image(for:)` and `thumbnailImage(for:)` return nil for a deleted image, which requires that the cache was cleared.